monkey/agent_plugins/exploiters/smb/src/smb_exploit_client.py

```diff
@@ -1,264 +1,226 @@
 import logging
-import ntpath
-from io import BytesIO
-from pathlib import PurePath
-from typing import Any, Dict, Optional, Sequence, Tuple
 
-from impacket.dcerpc.v5 import scmr, srvs, transport
-from impacket.smbconnection import SMBConnection
+# import ntpath
+from io import BytesIO
+from pathlib import PurePath, PureWindowsPath
+from typing import List, MutableSet, Optional, Tuple
 
 from common import OperatingSystem
 from common.credentials import Credentials
-from common.event_queue import IAgentEventPublisher
-from common.types import Event
-from infection_monkey.exploit.tools import RemoteFileCopyError
-from infection_monkey.exploit.tools.helpers import get_agent_dst_path
+from common.tags import (
+    T1021_ATTACK_TECHNIQUE_TAG,
+    T1105_ATTACK_TECHNIQUE_TAG,
+    T1110_ATTACK_TECHNIQUE_TAG,
+    T1135_ATTACK_TECHNIQUE_TAG,
+    T1210_ATTACK_TECHNIQUE_TAG,
+    T1569_ATTACK_TECHNIQUE_TAG,
+)
+from infection_monkey.exploit.tools import (
+    IRemoteAccessClient,
+    RemoteAuthenticationError,
+    RemoteCommandExecutionError,
+    RemoteFileCopyError,
+)
 from infection_monkey.i_puppet import TargetHost
 
-from .smb_command_builder import build_smb_command
+from .smb_client import ShareInfo, SMBClient
 from .smb_options import SMBOptions
-from .smb_utils import create_smb_connection, logout_guest, rpc_connect, smb_login
 
 logger = logging.getLogger(__name__)
 
 SERVICE_NAME = "InfectionMonkey"
+LOGIN_TAGS = {
+    T1021_ATTACK_TECHNIQUE_TAG,  # Remote Services
+    T1110_ATTACK_TECHNIQUE_TAG,  # Brute Force
+    T1210_ATTACK_TECHNIQUE_TAG,  # Exploitation of Remote Services
+}
+COPY_FILE_TAGS = {
+    T1021_ATTACK_TECHNIQUE_TAG,  # Remote Services
+    T1105_ATTACK_TECHNIQUE_TAG,  # Ingress Tool Transfer
+    T1135_ATTACK_TECHNIQUE_TAG,  # Network Share Discovery
+    T1210_ATTACK_TECHNIQUE_TAG,  # Exploitation of Remote Services
+}
+EXECUTION_TAGS = {
+    T1021_ATTACK_TECHNIQUE_TAG,  # Remote Services
+    T1210_ATTACK_TECHNIQUE_TAG,  # Exploitation of Remote Services
+    T1569_ATTACK_TECHNIQUE_TAG,  # Execution: System Services
+}
 
 
 class CopiedFileDetails:
     """Stores the details of a copied file"""
 
-    def __init__(self, remote_path: PurePath, destination_path: PurePath):
-        self.remote_path = remote_path
+    def __init__(self, destination_path: PurePath):
         self.destination_path = destination_path
 
 
-class SMBExploitClient:
+class SMBExploitClient(IRemoteAccessClient):
```

Comment:
What about renaming this to `SMBRemoteAccessClient`?

Reply:
I'm cool with that